rfcs/0011-separate-flyway-from-pipeline.md

```diff
@@ -0,0 +1,382 @@
+application will continue to run and process data until the schema migration completes.
+
+The new step that runs the Flyway migrations will be configured to run as a user that has the appropriate privileges to
+run migrations without having to elevate the privileges of the BFD Pipeline. This will allow schema migrations that
+create (non-user) database roles and grant privileges to run in Flyway.
+
+### Proposed Solution: Detailed Design
+[Proposed Solution: Detailed Design]: #proposed-solution-detailed-design
+
+Application design:
+
+* A new Java application 'bfd-db-migrator' will be introduced that performs these functions:
+  * Invokes Flyway (and thereby runs all pending migrations)
+  * Runs Hibernate validation against the BFD ORM (after Flyway finishes)
+  * Exits with a return code that indicates whether all operations were successful or not
+
+* Flyway invocations and hibernate validation will be removed from the BFD Pipeline and BFD API Server applications.
+
+* Test infrastructure will be altered to mimic the deployment by running the new application prior to starting the local
+BFD server for tests that run the BFD server as a standalone process.
+
+* Test infrastructure will be altered to invoke the new application logic via method call prior to running tests that
+require a BFD database but do not launch a BFD server.
+
+* The new Java application code will reside in the existing BFD repo as another sub-module of bfd-parent so that it will
+be built and tested as part of the current Github actions application workflow and Jenkins App build stage.
+
+Deployment design:
+
+![Proposed Deployment Sequence](./resources/0011-proposed-deployment-sequence.svg)
+
+<details><summary>Diagram Notes</summary>
+Like the previous diagram, this illustrates the deployment sequence with some broad definitions. This adds the `bfd-db-migrator` participant that includes a blocking deployment stage for each environment that must complete **before** proceeding to apply updates to the `bfd-pipeline` and `bfd-server` resources for each environment.
+</details>
+
+* A new migrator service account and associated migrator database role that has all required privileges for running
+migrations and hibernate validation will be created.
+
+* The Jenkins APP AMI build phase will build a new migrator AMI that includes the new application.
+
+* The migrator AMI and migrator service account will be used to run the migrator application on a new EC2 instance as
+part of a new Jenkins pipeline stage that will be inserted prior to each of the three stages that deploy to TEST,
+PROD-SBX, PROD.
+
+* The Jenkins pipeline will not proceed to the next stage (deploying the other applications) until the migrator
+application has exited with a status indicating success. In the event that this application status indicates that not
+all operations were successful, the Jenkins deployment will halt and be marked as a failure.
```

Comment:
Minor suggestion: You might note here that DB migrations are generally run transactionally, so this should not result in downtime.